app/models/concerns/translatable.rb

```diff
@@ -47,23 +47,69 @@ def attr_translatable(*attributes)
     end
   end
 
-  def get_translation(attribute)
-    translation = translations.find_by(translatable_attribute: attribute, locale: I18n.locale)
-    translation = get_translations(attribute).first  unless translation
-    return associated_translation(translation) if translation
-    nil
+  def get_translation(attribute, locale: I18n.locale, exact_match: false)
+    # important to filter against the association held in memory instead of performing a db lookup
+    # in order to redisplay potentially transient values
+    result = translations.find do |t|
+      t.translatable_attribute.to_sym == attribute.to_sym && t.locale.to_sym == locale.to_sym
+    end
+    unless result || exact_match
+      # fall back to first translations of any locale if desired locale not present
+      # (except when fetching in order to perform update)
+      result = translations.find do |t|
+        t.translatable_attribute.to_sym == attribute.to_sym
+      end
+    end
+    result
   end
 
   def get_translations(attribute)
     translations.where(translatable_attribute: attribute)
   end
 
-  def set_translation(attribute, text, locale: I18n.locale)
-    translation = get_translation(attribute)
+  def used_locales
+    result = []
+    if respond_to?(:division)
+      result = self.division.try(:resolve_default_locales)
+    end
+    result += translations.pluck(:locale)
+
+    # need to ignore any existing locales in UI if they don't currently correspond to a permitted locale
+    permitted = self.permitted_locales
+    result.map(&:to_sym).uniq.select{ |l| permitted.include?(l) }
+  end
+
+
+  # Beware, this is distinct from I18n.available_locales to flatten out the region specific locales
+  # for the purpose of the translatable fields.
+  # If we really want to support es-AR, etc in the translatable system, we need to tweak the migration
+  # and also think through all of the other implications
+  def permitted_locales
+    [:en, :es, :fr]   # todo: determine full list here
+    # Note, this list currently depends on being included in the I18n.available_locales list
+    # because we're leveraging the I18n system to translate our locale_name for the
+    # language selection drop-down.
+    # If we're okay with just using a simple map instead, then we could avoid that dependency.
+  end
+
+
+  def unused_locales
+    permitted_locales - used_locales
+  end
+
+
+
+  def delete_translation(attribute, locale)
+    translation = get_translation(attribute, locale:locale, exact_match: true)
+    translation.delete
+  end
+
+  def set_translation(attribute, text, locale: I18n.locale, old_locale: nil)
```

Comment:
A comment explaining the `old_locale` param would be nice. Not immediately clear what that's for.